Sources/BrowserServicesKit/Common/Extensions/URLExtension.swift

```diff
@@ -54,4 +54,44 @@ extension URL {
             password == nil
     }
 
+    // MARK: - Parameters
+
+    public enum ParameterError: Error {
+        case parsingFailed
+        case encodingFailed
+        case creatingFailed
+    }
+
+    public func addParameters(_ parameters: [String: String]) throws -> URL {
+        var url = self
+
+        for parameter in parameters {
+            url = try url.addParameter(name: parameter.key, value: parameter.value)
+        }
+
+        return url
+    }
+
+    public func addParameter(name: String, value: String) throws -> URL {
+        guard var components = URLComponents(url: self, resolvingAgainstBaseURL: false) else { throw ParameterError.parsingFailed }
+        var queryItems = components.queryItems ?? [URLQueryItem]()
+        let newQueryItem = URLQueryItem(name: name, value: value)
+        queryItems.append(newQueryItem)
+        components.queryItems = queryItems
+        guard let encodedQuery = components.percentEncodedQuery else { throw ParameterError.encodingFailed }
+        components.percentEncodedQuery = encodedQuery.encodingPluses()
+        guard let newUrl = components.url else { throw ParameterError.creatingFailed }
+        return newUrl
+    }
```

Comment:
We're [doing a similar thing on iOS](https://github.com/duckduckgo/iOS/blob/develop/Core/URLExtension.swift#L60), wonder if it's worth aligning these implementations to use on both. Mostly just thinking out loud, not worth being concerned about for this PR.

Reply:
Yeah so this was moved from the MacOS code, I think we can align the two implementations :)